cider-dap/src/main.rs

```diff
@@ -28,35 +30,59 @@ struct Opts {
 
 fn main() -> Result<(), MyAdapterError> {
     let opts: Opts = argh::from_env();
+    // Initializing logger
+    let log_path = "/tmp/output.log"; // Stores in tmp file for now, if testing, use a relative path
+    let file = OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .open(log_path)
+        .unwrap();
+
+    // Different decorators and drains for terminal and file logging -- async_drain picks the right one based on session
+    let term_decorator = slog_term::TermDecorator::new().build();
+    let file_decorator = slog_term::PlainDecorator::new(file);
+    let term_drain = slog_term::FullFormat::new(term_decorator).build().fuse();
+    let file_drain = slog_term::FullFormat::new(file_decorator).build().fuse();
+    let async_drain = if opts.is_multi_session {
+        slog_async::Async::new(term_drain).build().fuse()
+    } else {
+        slog_async::Async::new(file_drain).build().fuse()
```

Comment:
Move the initialization of the file_decorator and file drain into this arm rather than at the top